### PR TITLE
[improve][broker] make managedLedgerOffloadedReadPriority compatible with broker property

### PR DESCRIPTION
**Pull request by ethqunzhong**
### Motivation
As #16436 supported setManagedLedgerOffloadedReadPriority by broker.conf.
I add an unit-test to verify offloadpolicy-brokerconf-compatibled that test failed, show as follow:
image

offloadedDataReadPriority excepted value is `bookkeper-first` from broker's property but get default value `tiered-storage-first`.

### Modifications
Fix `OffloadPoliciesImpl#getCompatibleValue` logic: add broker's property `managedLedgerDataReadPriority` compatible.

### Does this pull request potentially affect one of the following parts:

_If `yes` was chosen, please highlight the changes_

-   Dependencies (does it add or upgrade a dependency): (no)
-   The public API: (no)
-   The schema: (no)
-   The default values of configurations: (no)
-   The wire protocol: (no)
-   The rest endpoints: (no)
-   The admin cli options: (no)
-   Anything that affects deployment: (no)
### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [x] `doc-not-needed` 
(Please explain why)
  
- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### Matching PR in forked repository
PR in forked repository: https://github.com/ethqunzhong/pulsar/pull/2

### CODE REVIEW

**Comment by ethqunzhong** [comment]
@Technoboy- PTAL.

File: pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/OffloadPoliciesImpl.java [hunk 1]
```diff
@@ -256,9 +257,9 @@ public void compatibleWithBrokerConfigFile(Properties properties) {
                     Long.parseLong(properties.getProperty(DELETION_LAG_NAME_IN_CONF_FILE)));
         }
 
-        if (properties.containsKey("managedLedgerDataReadPriority")) {
+        if (properties.containsKey(DATA_READ_PRIORITY_NAME_IN_CONF_FILE)) {
             setManagedLedgerOffloadedReadPriority(
```
Comment (HQebupt) [inline pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/OffloadPoliciesImpl.java:261]: This is better.
```
if (!properties.containsKey("managedLedgerOffloadedReadPriority")
                && properties.containsKey("DATA_READ_PRIORITY_NAME_IN_CONF_FILE")) {
            setManagedLedgerOffloadedReadPriority(
                    OffloadedReadPriority.fromString(properties.getProperty("DATA_READ_PRIORITY_NAME_IN_CONF_FILE")));
        }
```

Comment (ethqunzhong) [reply pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/OffloadPoliciesImpl.java:262]: Yes, Thanks. i had updated.

File: conf/broker.conf [hunk 2]
```diff
@@ -1133,6 +1133,10 @@ managedLedgerOffloadDeletionLagMs=14400000
 # (default is -1, which is disabled)
 managedLedgerOffloadAutoTriggerSizeThresholdBytes=-1
 
+# Read priority when ledgers exists in both bookkeeper and the second layer storage
```
Comment (ethqunzhong) [inline conf/broker.conf:1136]: i will add another doc PR to describe this new broker configuration.

**Comment by ethqunzhong** [comment]
@HQebupt @Technoboy- PTAL.

**Review by HQebupt — APPROVED** [review]

**Comment by ethqunzhong** [comment]
@hangc0276 ready to be merge after ci-test passed in my own repo PR. PTAL.

**Comment by ethqunzhong** [comment]
@Jason918 @zymap  PTAL.

**Comment by ethqunzhong** [comment]
i have add Matching PR in forked repository https://github.com/ethqunzhong/pulsar/pull/2
PTAL @Technoboy- @HQebupt @codelipenghui @Jason918 @zymap @hangc0276 @mattisonchao

**Review by HQebupt — APPROVED** [review]

**Comment by ethqunzhong** [comment]
CI ALL passed: https://github.com/ethqunzhong/pulsar/pull/2
PTAL, thanks for your help.
@codelipenghui @lhotari @eolivelli @merlimat

**Review by eolivelli — APPROVED** [review]

**Comment by eolivelli** [comment]
maybe you can rebase onto latest master

**Comment by ethqunzhong** [comment]
> maybe you can rebase onto latest master
thx your suggest. i had rebase onto latest master.

**Comment by ethqunzhong** [comment]
CI test failures related to Time elapsed failures while excute`deleteNamespaceGraceFully` , latest related pr is https://github.com/apache/pulsar/pull/17887

**Comment by ethqunzhong** [comment]
> CI test failures related to Time elapsed failures while excute`deleteNamespaceGraceFully` , latest related pr is #17887

@eolivelli it seem some slow flaky-test always failed with others CI all passed. could merge this pr into master and ignore this test failed ? 
thx your reply.

**Review by hangc0276 — APPROVED** [review]

**Pull request merged**